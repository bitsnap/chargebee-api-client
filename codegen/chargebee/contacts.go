package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListContacts generates chargebee client code to fetch all currencies
//
// API: https://{site}.chargebee.com/api/v2/contacts
//
// Documentation: https://apidocs.chargebee.com/docs/api/currencies?lang=curl#list_currencies
func GenerateListContacts() string {
	return client.GenerateList("Contacts", "Contact", "chargebee.com/api/v2/contacts")
}

// GenerateContactsModel generates chargebee contacts domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/contacts?prod_cat_ver=2#contact_attributes
func GenerateContactsModel() string {
	return client.GenerateModel("Contact", "Contact attributes", "https://apidocs.chargebee.com/docs/api/contacts")
}
