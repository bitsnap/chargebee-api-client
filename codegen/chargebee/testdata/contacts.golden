# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee contacts code generation when generating Chargebee API client should be able to generate contacts models" = '''
(string) (len=467) "type Contact struct {\n    Id string `json:\"id\" validate:\"required\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\" validate:\"required\"`\n    Phone string `json:\"phone\"`\n    Label string `json:\"label\"`\n    Enabled bool `json:\"enabled\" validate:\"required\"`\n    SendAccountEmail bool `json:\"send_account_email\" validate:\"required\"`\n    SendBillingEmail bool `json:\"send_billing_email\" validate:\"required\"`\n}"
'''
